controllers/datadogagent/override/container_test.go

```diff
@@ -7,54 +7,79 @@ package override
 
 import (
 	"fmt"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/common"
+	apiutils "github.com/DataDog/datadog-operator/apis/utils"
+	"k8s.io/apimachinery/pkg/api/resource"
 	"testing"
 
-	"github.com/DataDog/datadog-operator/apis/datadoghq/common"
 	commonv1 "github.com/DataDog/datadog-operator/apis/datadoghq/common/v1"
 	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
-	apiutils "github.com/DataDog/datadog-operator/apis/utils"
 	"github.com/DataDog/datadog-operator/controllers/datadogagent/feature/fake"
 	"github.com/stretchr/testify/assert"
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/resource"
 )
 
 func TestContainer(t *testing.T) {
-
+	agentContainer := &corev1.Container{
+		Name: string(commonv1.CoreAgentContainerName),
+	}
+	initVolContainer := &corev1.Container{
+		Name: string(commonv1.InitVolumeContainerName),
+	}
+	initConfigContainer := &corev1.Container{
+		Name: string(commonv1.InitConfigContainerName),
+	}
+	systemProbeContainer := &corev1.Container{
+		Name: string(commonv1.SystemProbeContainerName),
+	}
 	tests := []struct {
 		name            string
 		containerName   commonv1.AgentContainerName
 		existingManager func() *fake.PodTemplateManagers
 		override        v2alpha1.DatadogAgentGenericContainer
-		validateManager func(t *testing.T, manager *fake.PodTemplateManagers)
+		validateManager func(t *testing.T, manager *fake.PodTemplateManagers, containerName string)
 	}{
 		{
 			name:          "override container name",
 			containerName: commonv1.CoreAgentContainerName,
 			existingManager: func() *fake.PodTemplateManagers {
-				return fake.NewPodTemplateManagers(t)
+				return fake.NewPodTemplateManagers(t, corev1.PodTemplateSpec{
+					Spec: corev1.PodSpec{
+						Containers: []corev1.Container{*agentContainer},
+					},
+				})
 			},
 			override: v2alpha1.DatadogAgentGenericContainer{
 				Name: apiutils.NewStringPointer("my-container-name"),
 			},
-			validateManager: func(t *testing.T, manager *fake.PodTemplateManagers) {
+			validateManager: func(t *testing.T, manager *fake.PodTemplateManagers, containerName string) {
+				foundMatching := false
 				for _, container := range manager.PodTemplateSpec().Spec.Containers {
-					if container.Name == string(commonv1.CoreAgentContainerName) {
+					if container.Name == "my-container-name" {
+						foundMatching = true
 						assert.Equal(t, "my-container-name", container.Name)
+						break
 					}
 				}
+				if !foundMatching {
+					t.Errorf("Could not find matching container: %s", containerName)
```

Comment:
This isn't really an error right condition right, rather a failed unit test? If yes, probably better to use `assert.True(..)`